js/block.js

```diff
@@ -3656,14 +3651,27 @@ class Block {
                     '" />';
                 labelElem.classList.add("hasKeyboard");
                 this.label = docById("numberLabel");
+
+                // Add an input event listener to enforce the 6-digit limit
```

Comment:
How should we handle the minus (-) case? And decimal points?

Reply:
Upon checking the value of steps (which is a parameter of the doForward method) using console.log, I realized that the input for the forward block is determined by multiplying steps by 8, and this value is provided by the user. Initially, I apply Math.floor to this input, which rounds off decimal inputs. Then, I convert the rounded value to a string. The subsequent function checks the length of this string.

In the case of negative inputs, the "-" sign is also considered along with the digits when calculating the length of the input. For example, if the user types "-12345" as the input, the doForward block processes only the "-123" part of the input.

Let's consider some examples for clarity:

1) If you input "12345.67" into the forward block, it will return "1234" after the Math.floor operation. The value is then converted to a string, resulting in "12345". Since the length of this string is 5, we slice the first 4 digits and continue with the further procedure.

2) For negative numbers, if you input "-12345.789", after rounding off, it becomes "-12346". The length of "-12346" is 6, so we slice the first 4 digits, and "-123" is used for further processing.